### PR TITLE
Fix cgo issues causing a panic

### PR DESCRIPTION
**Pull request by jhchabran**
Follow-up to https://sourcegraph.slack.com/archives/C04MYFW01NV/p1687904137730639 where @unknwon spotted a failure in the pure-docker test suite, I was ablet to root-cause the issues down to the prom-wrapper binary being built with netgo but with cgo enabled, which led to a weird case, where it still tried to nslookup with cgo and segfaulted.

Passes the pure-docker test. See also long term fix: https://github.com/sourcegraph/sourcegraph/pull/54359

### Checklist

* [x] Sister [deploy-sourcegraph](https://github.com/sourcegraph/deploy-sourcegraph) change:
* [x] Sister [customer-replica](https://github.com/sourcegraph/deploy-sourcegraph-docker-customer-replica-1) change (if necessary, for any changes affecting pure-docker or configuration):
* [x] All images have a valid tag and SHA256 sum
### Test plan

CI is green

### CODE REVIEW

**Review by unknwon — APPROVED** [review]
Thank you!

**Comment by jhchabran** [comment]
@coury-clark I'm waiting for your final decision on how we want to approach this, with this hotfix or the backport at https://github.com/sourcegraph/sourcegraph/pull/54359

**Pull request closed without merging**
